sympy/stats/stochastic_process_types.py

```diff
@@ -580,18 +581,76 @@ def expectation(self, expr, condition=None, evaluate=True, **kwargs):
         raise NotImplementedError("Mechanism for handling (%s, %s) queries hasn't been "
                                 "implemented yet."%(expr, condition))
 
+
+class CommunicationClass(Basic):
+    def __new__(cls, states: list, recurrent: bool = None, period: int = None):
+        """Represents a single communication class for a Markov Chain.
+
+        Parameters
+        ==========
+
+        states : list
+            The list of integer states that are part of the equivalence class.
+        recurrent : bool, optional
+            Whether all the states in the class are recurrent or transient.
+        period: int, optional
+            The period of the states. None if it is transient.
+        """
+        # shamefully stolen from MutableDenseMatrix
+        self = Basic.__new__(cls)
+        self.states = states
+        self.recurrent = recurrent
+        self.period = period
```

Comment:
These two aren't needed anyways IMHO. The API for generating communication as simple as possible. This class can be removed for now.

Reply:
The periods are used within `limiting_dist()` to check if it should call `stationary_distribtion`

Recurrence is used in a lot of methods mainly for deciding which states to use for the decomposition, for the fundamental matrix and a few other things.

I also chose to keep them so that a user can type `X.communication_classes()` and then receive an output similar to what's at the bottom of [this pdf](https://drive.google.com/file/d/1mE9IgOdO-Nu9-xAaei74n28v5jVD7yvW/view?usp=drivesdk).